.github/workflows/release-integration.yml

```diff
@@ -91,18 +77,21 @@ jobs:
         with:
           image-ref: '${{ env.DOCKER_IMAGE_NAME }}:ci-scan'
           format: 'table'
-          exit-code: '0'
+          exit-code: '1'
           ignore-unfixed: true
           severity: 'CRITICAL,HIGH'
 
-      - name: Push versioned images
+      - name: Push pre-release images
+        if: ${{ github.event.release.prerelease }}
         run: |
+          echo "DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG}-pre" >> $GITHUB_ENV
```

Comment:
I know that you have tested in a fork but are we sure that every time you "touch" github's envfile the environment is instantaneously updated?

I had the feeling that is not happening and that is the reason I [used the environment variable directly.](https://github.com/newrelic/newrelic-k8s-metrics-adapter/pull/40/files#diff-2c3b5a007a3ca5391253871b413bf6d78192fe551b28bdfe115df67aa07528b1R78) and it is working:
 - From prod repo: https://hub.docker.com/layers/newrelic-k8s-metrics-adapter/newrelic/newrelic-k8s-metrics-adapter/0.3.0-pre/images/sha256-3436a0523cf0394b9e1f373976fb80436aa3f053d8b42eab4e9a39bac3411b2d?context=explore
 - From fork repo: https://hub.docker.com/r/jperezflorez123/newrelic-k8s-metrics-adapter/tags

Reply:
I think it does, a race condition like that would be horrible, but you are right, it is better to use classical Env Var since we are in the same step!